src/main/java/co/cask/cdap/guides/PageRankHandler.java

```diff
@@ -22,35 +22,40 @@
 import co.cask.cdap.api.service.http.HttpServiceRequest;
 import co.cask.cdap.api.service.http.HttpServiceResponder;
 import com.google.common.base.Charsets;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
 
 import java.net.HttpURLConnection;
-import javax.ws.rs.GET;
+import javax.ws.rs.POST;
 import javax.ws.rs.Path;
-import javax.ws.rs.QueryParam;
 
 /**
  * Handles queries retrieving the PageRank of a URL.
  */
-public class PageRankHandler extends AbstractHttpServiceHandler {
 
+public final class PageRankHandler extends AbstractHttpServiceHandler {
+
+  private static final Gson GSON = new Gson();
+  public static final String URL_KEY = "url";
   public static final String PAGE_RANKS_RANK_HANDLER = "pagerank";
 
   @UseDataSet(PageRankApp.PAGE_RANK_RANKS_DATASET)
-  private ObjectStore<Double> pageRanks;
+  private ObjectStore<Double> ranks;
 
   @Path(PAGE_RANKS_RANK_HANDLER)
-  @GET
-  public void getRank(HttpServiceRequest request, HttpServiceResponder responder, @QueryParam("url") String url) {
+  @POST
```

Comment:
Earlier we were using query param. UI does not support query parameter so we have to change it to POST since we cannot use path param as here we are querying with URL which will have / and : chars